progs/jubjub.py

```diff
@@ -226,3 +226,26 @@ def __str__(self):
 
     async def double(self):
         return self
+
+
+async def share_mul(context, p: Point, x: list) -> 'SharedPoint':
```

Comment:
I feel like this might be more appropriately placed in the `Point` class, so that we could do `multiplied = p.share_mul([x])` and it will work more natively with how the existing operations are structured.

Reply:
Also, the type signature would be better as `async def share_mul(context: Mpc, p: Point, x: list) -> SharedPoint`-- I only have the quotes in the other examples because you can't use a type signature of a class within the class.

Reply:
I was considering placing it in Point class, but the problem is the function needs the `SharedPoint` class which imports `Point` class itself.

Reply:
Oh, good point.

Reply:
BTW, you could actually put this in the `Point` class if you imported `SharedPoint` within the method-- 
```
async def share_mul(self, context, x: list) -> 'Point':
        """
        The multiplication of the share of a field element and a point
        e.g. [X] <- [x]G, where G is a point on the given elliptic curve
        x is the bitwise shared value,
        starting from the most significant bit.
        """
        from progs.jubjub import SharedPoint

        product = Ideal(self.curve)
        addend = self
        x_ = await gather(*[e.open() for e in reversed(x)])

        for i_ in x_:
            # Need equality application to compare if i == 1 here
            # Open it to compare for now
            if i_ == 1:
                product += addend
            addend = addend.double()

        return await SharedPoint.from_point(context, product)
```

That said, I'd be curious what @amiller thinks, as it would be putting mpc stuff outside of `progs`. Another thought is that we could add something so that you can easily call this on a `Point`--
```
async def share_mul(self, context, x: list) -> 'Point': 
    from progs.jubjub import share_mul
    return await share_mul(context, self, x)
```

Reply:
This can probably wait though, as this would make sense to throw in with a refactor of `mixins`